go/vt/vtgate/planbuilder/operators/route_planning.go

```diff
@@ -497,6 +497,14 @@ func tryMerge(
 		return nil, nil
 	}
 
+	if altARoute := aRoute.AlternateInKeyspace(bRoute.Keyspace); altARoute != nil {
```

Comment:
I don't understand this. do we always want to do this on both routes? shouldn't we just do this if they are not on the same keyspace? Maybe add a comment explaining this logic?